pkg/datarepair/checker/checker_test.go

```diff
@@ -21,57 +23,32 @@ func TestIdentifyInjuredSegments(t *testing.T) {
 		SatelliteCount: 1, StorageNodeCount: 4, UplinkCount: 0,
 	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
 		checker := planet.Satellites[0].Repair.Checker
-		checker.Interval.Stop()
+		checker.Loop.Stop()
 
-		const numberOfNodes = 10
-		pieces := make([]*pb.RemotePiece, 0, numberOfNodes)
-		// use online nodes
-		for i, storagenode := range planet.StorageNodes {
-			pieces = append(pieces, &pb.RemotePiece{
-				PieceNum: int32(i),
-				NodeId:   storagenode.Identity.ID,
-			})
-		}
-
-		// simulate offline nodes
-		expectedLostPieces := make(map[int32]bool)
-		for i := len(pieces); i < numberOfNodes; i++ {
-			pieces = append(pieces, &pb.RemotePiece{
-				PieceNum: int32(i),
-				NodeId:   storj.NodeID{byte(i)},
-			})
-			expectedLostPieces[int32(i)] = true
+		//add noise to pointerdb before bad record
+		for x := 0; x < 1000; x++ {
+			makePointer(t, planet, fmt.Sprintf("a-%d", x), false)
 		}
-		pointer := &pb.Pointer{
-			Remote: &pb.RemoteSegment{
-				Redundancy: &pb.RedundancyScheme{
-					MinReq:          int32(4),
-					RepairThreshold: int32(8),
-				},
-				PieceId:      "fake-piece-id",
-				RemotePieces: pieces,
-			},
+		//create piece that needs repair
+		makePointer(t, planet, fmt.Sprintf("b"), true)
+		//add more noise to pointerdb after bad record
+		for x := 0; x < 1000; x++ {
+			makePointer(t, planet, fmt.Sprintf("c-%d", x), false)
 		}
-
-		// put test pointer to db
-		pointerdb := planet.Satellites[0].Metainfo.Service
-		err := pointerdb.Put(pointer.Remote.PieceId, pointer)
-		assert.NoError(t, err)
-
-		err = checker.IdentifyInjuredSegments(ctx)
+		err := checker.IdentifyInjuredSegments(ctx)
 		assert.NoError(t, err)
 
 		//check if the expected segments were added to the queue
 		repairQueue := planet.Satellites[0].DB.RepairQueue()
 		injuredSegment, err := repairQueue.Dequeue(ctx)
 		assert.NoError(t, err)
 
-		assert.Equal(t, "fake-piece-id", injuredSegment.Path)
-		assert.Equal(t, len(expectedLostPieces), len(injuredSegment.LostPieces))
+		numValidNode := int32(len(planet.StorageNodes))
+		assert.Equal(t, "b", injuredSegment.Path)
+		assert.Equal(t, len(planet.StorageNodes), len(injuredSegment.LostPieces))
 		for _, lostPiece := range injuredSegment.LostPieces {
-			if !expectedLostPieces[lostPiece] {
-				t.Error("should be lost: ", lostPiece)
-			}
+			assert.True(t, lostPiece >= numValidNode, fmt.Sprintf("%d >= %d \n", lostPiece, numValidNode))
+			assert.True(t, lostPiece < numValidNode*2, fmt.Sprintf("%d < %d \n", lostPiece, numValidNode*2))
```

Comment:
Can you add comments (or error message) why such conditions? Its a little bit hard to understand with just numbers in fail info.

Reply:
done